### ISSUE TITLE
bootkube-start does not complete

### ISSUE THREAD

**Issue by rubda** [poseidon/typhoon#123]
### Environment

* Platform: aws
* OS: container-linux
* Terraform: v0.11.3
* Plugins: 
   + provider.aws v1.5.0
   + provider.ct (unversioned)
   + provider.local v1.1.0
   + provider.null v1.0.0
   + provider.template v1.0.0
   + provider.tls v1.0.1
   
### Problem

bootkube-start is taking forever to complete, not sure if it will even complete. The last time I got it working was yesterday, but today I have tried multiple times with no success. 

```
module.MODULENAME.null_resource.bootkube-start: Still creating... (1h2m0s elapsed)
```

### Desired Behavior

To provision a Kubernetes cluster on AWS with one controller and two workers within 30 minutes at least. According to the [docs](https://typhoon.psdn.io/topics/performance/) I should be up and running within 10 minutes.

### Steps to Reproduce

Basically I just followed these [steps](https://typhoon.psdn.io/aws/).

AWS credentials are working, DNS Zone is added, terraform-provider-ct is installed and the ssh key in my module is also added to my ssh-agent. Worth mentioning is that I have successfully provisioned clusters on DigitalOcean several times, but seem to have the same problem over there as well now.

**Comment by dghubble** [comment]
Reports I've seen like this boils down to some `ssh-agent` problem, misconfiguration, or having a controller instance type that is too small or somehow failing health checks.

Although you said you added your SSH public key to `ssh-agent`, can you be sure:

* `ssh-agent` lists your key
* the key corresponds to the `ssh_authorized_key`
* you can SSH to a controller yourself (via IP as a sanity check, then via `{cluster_name}.${dns_zone}`)

Check the AWS dashboard shows the ELB with healthy backends.

rel: #122

**Comment by dghubble** [comment]
I have this suspicion that you may be the person that was modifying the VPC setup, causing DNS to fail, so kubelet image fetches will fail, so the health checks will fail, and there will be no healthy controllers to SSH into. Or am I thinking of someone else?

**Comment by rubda** [comment]
From what I can see my ssh-agent lists my key and the key is the one that corresponds to the ssh_authorized_key. 

I can SSH to the controller (I got the ```public_ip``` from using ```terraform show``` and retrieving it under ```module.MODULENAME.aws_instance.controllers```). 

However I can't SSH to ```{cluster_name}.${dns_zone})``` and when pinging it I get an ip address that does not match the one I used in the previous step. The connection is refused when trying to SSH.

Regarding your last post, since I don't know what you're meaning I suppose I'm not that person. I have basically used the same provider and module file shown in the examples. I've switched out the obvious variables that need to be my own personal ones.

**Comment by dghubble** [comment]
Ok. Well the SSH step you're stuck at is sshing to `${cluster_name}.${dns_zone}`. Its a DNS alias record resolving to the ELB for controllers. So you'll want to figure out why that doesn't work. Check the alias record (shown as two A records via dig, not the actual instance IPs). Check the ELB health.

**Comment by rubda** [comment]
Recreated my DNS Zone and readjusted my name servers, got it working now. Not sure why it was causing problems. Thanks for the help!
